docs/pages/technical/network.rst

```diff
@@ -0,0 +1,77 @@
+.. _nuts-network-technical:
+
+Nuts Network engine
+###################
+
+This section describes the workings of the Nuts Network engine.
+
+Status
+******
+
+This engine intents to provide a decentralized, persistent ledger for Nuts nodes to share information. While it is currently
+Work-in-Progress, we intent to migrate the Registry Github-sync mechanism and Corda consents to it. These are known as
+*applications* on the network. The Registry application will be realized first and will serve as Proof-of-Concept (POC) for the
+network as a viable distributed ledger for Nuts. If successful we can decide whether to migrate Corda consents to the network too.
+
+The Registry application will first operate alongside the Registry's Github-sync as to assess learn how the networks behaves
+in real-world scenarios and to assess the POC. If successful we can decide whether switch from Github-sync to the network application.
+
+Design
+******
+
+The business end of the engine consists of 5 layers:
+
+.. raw:: html
+    :file: ../../_static/images/network_engine_layers.svg
+
+Applications
+^^^^^^^^^^^^
+
+When services want to exchange information with other Nuts nodes they can build their application on top of the *DocumentLog* layer.
+Internal (to the network engine) applications can feed back information into one of the lower layers: the *NodeList* application
+is used to discover new potential peers for the P2P layer.
+
+Interfaces
+^^^^^^^^^^
+
+The Network engine has 2 interfaces:
+
+1. REST interface for diagnostics (like all engines) and for applications to query/add documents.
+2. Message queue for applications to subscribe on document events.
+
+Applications subscribe to specific document types, e.g. **nuts.node-info** for documents produced/consumed by the NodeList.
+
+DocumentLog
+^^^^^^^^^^^
+
+The DocumentLog provides a persistent document log which contains all documents on the network (known by the local node),
```

Comment:
maybe a disclaimer that in the future the log will consist of hashes and that the actual data has to be fetched. Both handled by the DocumentLog, but still a distinction

Reply:
Done